cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -150,3 +206,56 @@ func (crs *containerRegistryService) createRegistriesClient(
 
 	return client, nil
 }
+
+func (crs *containerRegistryService) getAcrToken(
+	ctx context.Context,
+	subscriptionId string,
+	loginServer string,
+) (*acrToken, error) {
+	creds, err := crs.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("getting credentials for subscription '%s': %w", subscriptionId, err)
+	}
+
+	token, err := creds.GetToken(ctx, policy.TokenRequestOptions{Scopes: auth.LoginScopes})
+	if err != nil {
+		return nil, fmt.Errorf("getting token for subscription '%s': %w", subscriptionId, err)
+	}
+
+	formData := url.Values{}
+	formData.Set("grant_type", "access_token")
+	formData.Set("service", loginServer)
```

Comment:
Do you think there's some upfront validation we should be adding to ensure `loginServer` follows a desired format of: `contosoregistry.azurecr.io`? I see that this value is actually user-provided in some cases, so we should probably ensure that this is well-documented and well-validated.

Reply:
We may also benefit from consistency in naming. `registryName` or `loginServerName`